### PR TITLE
fix: panic in vmss cache conversion

### PR DESCRIPTION
**Pull request by andyzhangx**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
fix: panic in cache conversion
This PR also check all cache conversion places, and fix panic if there is nil cache.

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

 - current nil cache conversion panic:
 
<details>

```
panic: interface conversion: interface {} is nil, not *sync.Map
 
goroutine 79 [running]:
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*ScaleSet).updateCache(0xc0004390e0?, {0xc0004390e0, 0x21}, {0xc0005fc7c3, 0x2d}, {0xc0007df8a0, 0x1b}, {0xc000f510fc, 0x4}, 0xc0000b4f50)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_vmss_cache.go:334 +0x7fb
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*ScaleSet).WaitForUpdateResult.func1()
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_controller_vmss.go:141 +0xca
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*ScaleSet).WaitForUpdateResult(0xc0003ff4a0?, {0x1e6acf8?, 0xc0003ff4a0?}, 0xc000f39860?, {0xc0004390e0?, 0xc000d34b00?}, {0xc0005fc7c3?, 0x160?}, {0x1bc2561, 0xb})
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_controller_vmss.go:149 +0x17e
sigs.k8s.io/cloud-provider-azure/pkg/provider.(*controllerCommon).AttachDisk(0xc000400000, {0x1e6acf8, 0xc0003ff4a0}, 0x0, {0xc0005fc813, 0x4f}, {0xc0005fc780, 0xe2}, {0xc0004390e0, 0x21}, ...)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/sigs.k8s.io/cloud-provider-azure/pkg/provider/azure_controller_common.go:274 +0x1202
sigs.k8s.io/azuredisk-csi-driver/pkg/azuredisk.(*Driver).ControllerPublishVolume(0xc000162180, {0x1e6acf8, 0xc0003ff4a0}, 0xc0004136e0)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/pkg/azuredisk/controllerserver.go:403 +0xec6
github.com/container-storage-interface/spec/lib/go/csi._Controller_ControllerPublishVolume_Handler.func1({0x1e6acf8, 0xc0003ff4a0}, {0x1ad1e20?, 0xc0004136e0})
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:5712 +0x78
sigs.k8s.io/azuredisk-csi-driver/pkg/csi-common.logGRPC({0x1e6acf8, 0xc0003ff4a0}, {0x1ad1e20?, 0xc0004136e0?}, 0xc0002f9820, 0xc00042cdb0)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/pkg/csi-common/utils.go:80 +0x35e
github.com/container-storage-interface/spec/lib/go/csi._Controller_ControllerPublishVolume_Handler({0x1ba25a0?, 0xc000162180}, {0x1e6acf8, 0xc0003ff4a0}, 0xc000413560, 0x1cbe9b8)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/github.com/container-storage-interface/spec/lib/go/csi/csi.pb.go:5714 +0x138
google.golang.org/grpc.(*Server).processUnaryRPC(0xc000464a80, {0x1e70998, 0xc0002b2680}, 0xc00011b440, 0xc0003fefc0, 0x2b37db0, 0x0)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:1283 +0xcfe
google.golang.org/grpc.(*Server).handleStream(0xc000464a80, {0x1e70998, 0xc0002b2680}, 0xc00011b440, 0x0)
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:1620 +0xa2f
google.golang.org/grpc.(*Server).serveStreams.func1.2()
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:922 +0x98
created by google.golang.org/grpc.(*Server).serveStreams.func1
    /go/src/github.com/kubernetes-sigs/azuredisk-csi-driver/vendor/google.golang.org/grpc/server.go:920 +0x28a
```

</details>

#### Does this PR introduce a user-facing change?

```release-note
fix: panic in cache conversion
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs
fix: panic in cache conversion
```

### CODE REVIEW

**Comment by MartinForReal** [comment]
Do you think we should fix in cache side instead？

File: pkg/provider/azure_controller_vmss.go [hunk 1]
```diff
@@ -223,7 +223,8 @@ func (ss *ScaleSet) DetachDisk(ctx context.Context, nodeName types.NodeName, dis
 		// Update the cache with the updated result only if its not nil
 		// and contains the VirtualMachineScaleSetVMProperties
 		if updateResult != nil && updateResult.VirtualMachineScaleSetVMProperties != nil {
-			if updErr := ss.updateCache(vmName, nodeResourceGroup, vm.VMSSName, vm.InstanceID, updateResult); updErr != nil {
+			if err := ss.updateCache(vmName, nodeResourceGroup, vm.VMSSName, vm.InstanceID, updateResult); err != nil {
+				klog.Errorf("updateCache(%s, %s, %s) failed with error: %v", vmName, nodeResourceGroup, vm.VMSSName, vm.InstanceID, err)
```
Comment (andyliuliming) [inline pkg/provider/azure_controller_vmss.go:227]: %s instead %v for err

File: pkg/provider/azure_vmss.go [hunk 2]
```diff
@@ -1690,6 +1702,9 @@ func (ss *ScaleSet) ensureBackendPoolDeletedFromVMSS(backendPoolID, vmSetName st
 			klog.Errorf("ensureBackendPoolDeletedFromVMSS: failed to get vmss uniform from cache: %v", err)
 			return err
 		}
+		if cachedUniform == nil {
```
Comment (andyliuliming) [inline pkg/provider/azure_vmss.go:1705]: no unit test for this package? :)

**Comment by andyzhangx** [comment]
> Do you think we should fix in cache side instead？E.g. add an extra return value for get function

@MartinForReal if cache is nil, then return nil pointer error, I think that would make the PR more clear.

**Comment by andyzhangx** [comment]
> > Do you think we should fix in cache side instead？E.g. add an extra return value for get function
> 
> @MartinForReal if cache is nil, then return nil pointer error, I think that would make the PR more clear.

@MartinForReal actually we allow nil cache in timedCache functions:
https://github.com/kubernetes-sigs/cloud-provider-azure/blob/3aa5adafabd1961dda9acbd09b9912d25ffe6eb8/pkg/cache/azure_cache_test.go#L79-L82

If we return nil cache error in timedCache Get function, that would make fundamental changes, I tend to only fix this issue in csi driver related functions, that would be safer.

File: pkg/provider/azure_controller_vmss.go (hunk 1 above)
Comment (andyzhangx) [reply pkg/provider/azure_controller_vmss.go:227]: what's the diff? we use %v for printing err everywhere in this repo

**Pull request merged**